element/element.go

```diff
@@ -0,0 +1,36 @@
+package element
```

Comment:
the name of this package isn't very illuminating.  Maybe just "matcher"?

Reply:
I split `element` package out from `matcher` because I was getting circular dependency issues with how everything is currently implemented and was trying to minimize how much refactoring was done.